### PR TITLE
Communication - Refactor Chat Tests to use StepVerifier

### PR DESCRIPTION
**Pull request by minnieliu**

### CODE REVIEW

**Comment by jbeauregardb** [comment]
LGTM 👍🏻

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java [hunk 1]
```diff
@@ -57,92 +60,137 @@ protected void afterTest() {
...
         assertNotNull(chatThreadClient);
         assertEquals(chatThreadClient.getChatThreadId(), threadId);
     }
 
     @Test
     public void canGetExistingChatThread() {
+        // Arrange
         CreateChatThreadOptions threadRequest = ChatOptionsProvider.createThreadOptions(
             firstThreadMember.getId(), secondThreadMember.getId());
         ChatThreadAsyncClient chatThreadClient = client.createChatThread(threadRequest).block();
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java:110]: still need to remove this block. It's what I was asking Srikanta about during the tutorial. Let me find his answer about how to do it

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadAsyncClientTest.java [hunk 2]
```diff
@@ -64,158 +67,219 @@ protected void afterTest() {
...
+            })
+            .verifyComplete();
     }
 
     @Test
     public void canDeleteExistingMessage() {
+        // Arrange
         SendChatMessageOptions messageRequest = ChatOptionsProvider.sendMessageOptions();
-
         SendChatMessageResult response = chatThreadClient.sendMessage(messageRequest).block();
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadAsyncClientTest.java:216]: Same comment here, need to get rid of this .block() too

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java (hunk 1 above)
Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java:110]: "What if we need to set up/create an object on the service first (as part of the arrange), before acting on it? Do we initialize the sync client too, and use it to arrange the setup? Or do we use a multi…"

you can chain multiple service calls asynchronously through Reactor operators like then().. so, the first call will do the necessary setup on the service and then call the next method that you want to test

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadAsyncClientTest.java (hunk 2 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadAsyncClientTest.java]: Done

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java]: Done

**Review by jbeauregardb — APPROVED** [review]

**Review by angiurgiu — APPROVED** [review]
Looks good now. Thanks for improving the tests!

**Pull request merged**